core/metacling/src/TClingMethodInfo.cxx

```diff
@@ -82,6 +82,48 @@ class TClingMethodInfo::SpecIterator
    Iterator fEnd;
 };
 
+class TClingMethodInfo::UsingIterator
+{
+public:
+   typedef clang::UsingDecl::shadow_iterator Iterator;
+
+   UsingIterator(cling::Interpreter* interp, Iterator begin, Iterator end) : fInterp(interp), fIter(begin), fEnd(end) {}
+   explicit UsingIterator(cling::Interpreter* interp, clang::UsingDecl *decl) :
+       fInterp(interp), fIter(decl->shadow_begin()), fEnd(decl->shadow_end()) {}
+
+   FunctionDecl *operator* () const {
+      clang::ConstructorUsingShadowDecl* shadow_ctor = llvm::dyn_cast<clang::ConstructorUsingShadowDecl>(*fIter);
+      if (shadow_ctor) {
```

Comment:
`if (auto shadow_ctor = llvm::dyn_cast<const clang::ConstructorUsingShadowDecl>(*fIter)) {`
reduces the scope of `shadow_ctor`.